ui/src/app/apps/apps/apps.component.html

```diff
@@ -0,0 +1,190 @@
+<div [ngBusy]="busy" id="applications-list">
+
+  <div class="heading">
+    <div class="actions" [appRoles]="['ROLE_CREATE']">
+      <div class="dropdown" dropdown id="add">
+        <button dropdownToggle type="button" class="btn btn-primary btn-dropdown" data-toggle="dropdown"
+                aria-haspopup="true"
+                aria-expanded="false">
+          Add Application(s)
+          <span class="caret"></span>
+        </button>
+        <ul *dropdownMenu class="dropdown-menu dropdown-menu-right">
+          <li><a (click)="registerApps()">Register application(s)</a></li>
+          <li><a (click)="bulkImportApps()">Bulk import application(s)</a></li>
+        </ul>
+      </div>
+    </div>
+    <div class="description">
+      <h1>Apps</h1>
+      <p>
+        This section lists all the available applications and provides the control to register/unregister them (if
+        applicable).
+      </p>
+    </div>
+  </div>
+
+  <div id="filters" class="pui-filter" *ngIf="!isAppsEmpty()">
+    <form #registerAppsForm="ngForm" name="registerAppsForm" role="form" (ngSubmit)="search()" novalidate class="row">
+      <div class="col-md-8">
+        <div id="dropdown-actions" class="dropdown" *ngIf="countSelected() > 0" dropdown>
+          <button dropdownToggle type="button" class="btn btn-default btn-dropdown" data-toggle="dropdown"
+                  aria-haspopup="true"
+                  aria-expanded="false">
+            Actions&nbsp;
+            <span *ngIf="countSelected() == 1">(1)</span>
+            <span *ngIf="countSelected() > 1">({{ countSelected() }})</span>
+            <span class="caret"></span>
+          </button>
+          <ul *dropdownMenu class="dropdown-menu">
+            <li><a id="unregister-apps" href="#" (click)="unregisterAppsSelected()">Unregister application(s)</a></li>
+          </ul>
+        </div>
+      </div>
+      <div class="col-md-8">
+        <div class="input-group input-group-filter">
+          <input type="text" id="q" name="q" class="form-control sm" placeholder="Filter items" [(ngModel)]="form.q">
+          <div dropdown class="filter-dropdown">
+            <a class="filter-dropdown-toggle" dropdownToggle style="min-width: 170px;">
+              <span *ngIf="form.type == '' || form.type == null">
+                All types
+              </span>
+              <span *ngIf="form.type != '' && form.type!== null">
+                Type: <strong>{{ form.type }}</strong>
+              </span>
+              <span class="caret"></span>
+            </a>
+            <ul class="dropdown-menu dropdown-menu-right" *dropdownMenu="">
+              <li [class.active]="form.type == '' || form.type == null"><a (click)="form.type = ''">All type</a></li>
```

Comment:
Change `All type` to `All types`

![image](https://user-images.githubusercontent.com/685363/36179119-3a7aaf8c-10bf-11e8-98a7-7951fad06713.png)

Also, we may consider refactoring the `ApplicationType` domain class. Theoretically, it has all the relevant data to populate the drop-down data. BUT admittedly Typescript enums are "underwhelming".

We may consider using something like: https://github.com/LMFinney/ts-enums

Reply:
Hard coded - Have to be update with a new component lib